app/controllers/characters_controller.rb

```diff
@@ -69,6 +69,19 @@ def update
     end
   end
 
+  def duplicate
+    @dup = @character.dup
```

Comment:
(I've wrapped it in `Character.transaction` anyway but it doesn't detect if there's a problem. Similarly to how `characters#do_replace` doesn't seem to.)